Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/impl/RequestsManager.java

```diff
@@ -19,16 +29,85 @@
     /* (non-Javadoc)
      * @see edu.asu.conceptpower.app.core.IRequestsManager#addReviewRequest(edu.asu.conceptpower.core.ReviewRequest)
      */
-    public void addReviewRequest(ReviewRequest newReviewRequest) {
+    public void addReviewRequest(ReviewRequest reviewRequest) {
+        reviewRequest.setId(generateId(REVIEW_PREFIX));
+        dbClient.store(reviewRequest);
+    }
+    
+    /* (non-Javadoc)
+     * @see edu.asu.conceptpower.app.core.IRequestsManager#getLatestReview(java.lang.String)
+     */
+    public ReviewRequest getLatestReview(String conceptId) {
+        List<ReviewRequest> reviewRequests = getAllReviews(conceptId);
+       
+        if(reviewRequests!= null && reviewRequests.size() > 0) {
+            return reviewRequests.get(reviewRequests.size() - 1);
+        }
         
-        dbClient.store(newReviewRequest);
+        return null;
     }
     
     /* (non-Javadoc)
-     * @see edu.asu.conceptpower.app.core.IRequestsManager#getReview(java.lang.String)
+     * @see edu.asu.conceptpower.app.core.IRequestsManager#getAllReviews(java.lang.String)
      */
-    public ReviewRequest getReview(String conceptId) {
+    public List<ReviewRequest> getAllReviews(String conceptId){
+        List<ReviewRequest> reviewRequests = dbClient.getAllReviews(conceptId);
         
-        return dbClient.getReviewRequestForConcept(conceptId);
+        if(reviewRequests!= null && reviewRequests.size() > 0) {
+            reviewRequests = new ArrayList<>(reviewRequests);
+            Collections.sort(reviewRequests, Comparator.comparing(ReviewRequest :: getCreatedAt, Comparator.nullsFirst(Comparator.naturalOrder())));
+            return reviewRequests;
+        }
+        
+        return null;
+    }
+    
+    /* (non-Javadoc)
+     * @see edu.asu.conceptpower.app.core.IRequestsManager#updateReview(edu.asu.conceptpower.core.ReviewRequest)
+     */
+    
+    public ReviewRequest updateReview(String reviewId, ReviewStatus reviewStatus, Comment comment, OffsetDateTime createdAt,String updatedBy) {  
+        ReviewRequest storedRequest = dbClient.getReview(reviewId);
```

Comment:
Can reviewId be null?

Reply:
Good question. For all the null cases you've asked, there is a possibility because it's coming from the frontend. The database client handles that. We are checking the database response and making necessary changes.